alternative-routes/index.js

```diff
@@ -115,39 +119,67 @@ const decodePolylines = (route, alternatives) => {
 };
 
 /**
- * Show journey specific information like duration, consumption, costs etc.
- *
- * @param data {object} route specification
+ * Small function that sets the time on how much longer the different routes are
+ * @param { object } route - The fastest route
+ * @param { array } alternatives - The alternative route objects
  */
-export const displayRouteData = data => {
-  if (document.querySelector('.tags').style.display !== 'flex') {
-    document.getElementById('loader').remove();
-    document.querySelector('.tags').style.display = 'flex';
-  }
-
-  // the total duration of the journey (including charge time), in seconds
-  document.getElementById('duration').innerHTML = `${getDurationString(data.duration ?? 0)}`;
-
-  // the total distance of the route, in meters
-  document.getElementById('distance').innerHTML = data.distance ? `${(data.distance / 1000).toFixed(0)} km` : 'Unknown';
-
-  // the amount of stops in this route
-  document.getElementById('stop').innerHTML = `${data.charges ?? 0} stops`;
+const renderTabData = (route, alternatives) => {
+  const routeDurations = [
+    'Fastest',
+    `+${getDurationString(alternatives[0].duration - route.duration)}`,
+    `+${getDurationString(alternatives[1].duration - route.duration)}`,
+  ];
+
+  document.querySelectorAll('.tab').forEach((tab, idx) => {
+    tab.lastElementChild.innerHTML = routeDurations[idx];
+  });
+};
 
-  // the total time required to charge of the entire route, in seconds
-  document.getElementById('charge-duration').innerHTML = getDurationString(data.chargeTime ?? 0);
+/**
+ * Function that renders the header details
+ * @param { object } data - All available route data
+ */
+export const renderRouteHeader = data => {
+  const routeDistance = data.distance ? `${(data.distance / 1000).toFixed(0)} km` : 'Unknown';
+  const routeStops = `${data.charges ?? 0} stops`;
+  const routeEnergy = data.consumption ? `${data.consumption.toFixed(2)} kWh` : 'Unknown';
 
-  // the total energy used of the route, in kWh
-  document.getElementById('consumption-overview').innerHTML = data.consumption
-    ? `${data.consumption.toFixed(2)} kWh`
-    : 'Unknown';
-  document.getElementById('consumption').innerHTML = data.consumption
-    ? `${data.consumption.toFixed(2)} kWh`
-    : 'Unknown';
+  document.getElementById('duration').innerHTML = `${getDurationString(data.duration ?? 0)}`;
+  document.getElementById('route-metadata').innerHTML = `${routeDistance} / ${routeStops} / ${routeEnergy}`;
+};
```

Comment:
The route header should also be updated when the user switches to an alternative route.

Reply:
Yup, missed that one. Good catch.